### PR TITLE
CAMEL-14268: Component auto configuration fails in Java 11

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java [hunk 1]
```diff
@@ -516,6 +516,32 @@ private static boolean doBindProperties(CamelContext camelContext, Object target
             uOptionPrefix = optionPrefix.toUpperCase(Locale.US);
         }
 
+        // we need to set references first
```
Comment (davsclaus) [inline core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java]: Yeah that seems like a good idea as they are likely more complex objects than simple types for boolean, ints, strings etc. Then having those set first then chances are the others can be set, also if they are sub options on these objects that was set as reference parameters. Also when the order of properties are not correct, maybe the end user didnt set it right, so this is an improvement overall anyway.

**Comment by lburgazzoli** [comment]
let me work on a more advanced implementation

**Comment by lburgazzoli** [comment]
@davsclaus  @oscerd mind doing an additional review ?

**Comment by davsclaus** [comment]
Yeah looks good, I would maybe add a little more code comment why we sort by nested deep with the count of dots.

**Comment by lburgazzoli** [comment]
will do

**Comment by lburgazzoli** [comment]
@davsclaus should be ok now

**Comment by davsclaus** [comment]
Super work

**Comment by lburgazzoli** [comment]
should be backported ?

**Comment by davsclaus** [comment]
Yeah that would be good for java 11 users of camel 3.0.x

**Comment by lburgazzoli** [comment]
done

**Pull request merged**